File: scratch/inspect_xlsx.py
import openpyxl
import os

def inspect_file(filepath):
    print(f"=== Inspecting {os.path.basename(filepath)} ===")
    wb = openpyxl.load_workbook(filepath, read_only=True)
    print("Sheets:", wb.sheetnames)
    for sheetname in wb.sheetnames:
        sheet = wb[sheetname]
        print(f"  Sheet: {sheetname}")
        # Print first 5 rows
        for i, row in enumerate(sheet.iter_rows(values_only=True)):
            if i >= 5:
                break
            # print non-empty columns
            row_vals = [str(x)[:30] for x in row if x is not None]
            if row_vals:
                print(f"    Row {i+1}: {row_vals[:8]}")

files = [
    r'd:\HOUSEHOLDSURVEY\Staff and Event details - CARE.xlsx',
    r'd:\HOUSEHOLDSURVEY\H.Survey Master File - CARE.xlsx'
]

for f in files:
    if os.path.exists(f):
        inspect_file(f)
    else:
        print(f"File not found: {f}")

File: scratch/inspect_xlsx.js
const XLSX = require('xlsx');
const path = require('path');
const fs = require('fs');

function inspectFile(filepath) {
  console.log(`=== Inspecting ${path.basename(filepath)} ===`);
  const workbook = XLSX.readFile(filepath);
  console.log("Sheets:", workbook.SheetNames);
  workbook.SheetNames.forEach(sheetname => {
    console.log(`  Sheet: ${sheetname}`);
    const sheet = workbook.Sheets[sheetname];
    const rows = XLSX.utils.sheet_to_json(sheet, { header: 1 });
    // Print first 5 rows
    for (let i = 0; i < Math.min(rows.length, 5); i++) {
      const row = rows[i];
      if (row && row.length > 0) {
        console.log(`    Row ${i+1}:`, row.slice(0, 8).map(x => String(x).substring(0, 30)));
      }
    }
  });
}

const files = [
  path.join(__dirname, '../Staff and Event details - CARE.xlsx'),
  path.join(__dirname, '../H.Survey Master File - CARE.xlsx')
];

files.forEach(f => {
  if (fs.existsSync(f)) {
    inspectFile(f);
  } else {
    console.log(`File not found: ${f}`);
  }
});
